### PR TITLE
Adds Client

### PR DESCRIPTION
**Pull request by kpp**

### CODE REVIEW

File: example/Client.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+#include <stdio.h>
+
+#include <Client.h>
+#include <Game.h>
+#include <BwString.h>
+#include <Iterator.h>
+#include <Player.h>
+
+static Game* BWGame;
+static Client* BWClient;
```
Comment (0x7CFE) [inline example/Client.c]: `= 0`

File: example/Client.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    while ( !Game_isInGame(BWGame) ) {
+        Client_update(BWClient);
+        reconnect();
+    }
+
+    fprintf(stderr, "Starting match\n");
+    Game_sendText(BWGame, "Hello world!");
+
+    BwString* map_name = Game_mapName(BWGame);
```
Comment (0x7CFE) [inline example/Client.c]: `const`

File: example/Client.c [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    Game_sendText(BWGame, "The map is %s!", BwString_data(map_name));
```
Comment (0x7CFE) [inline example/Client.c]: Better enclose `%s` in quotes within the string.

File: example/Client.c (hunk 2 above)
Comment (0x7CFE) [inline example/Client.c]: `assert(map_name)`

File: example/Client.c [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        Game_sendText(BWGame, "The following players are in this replay:");
+        Iterator* const players = (Iterator*) Game_getPlayers(BWGame);
+        assert(players);
+
+        for (; Iterator_valid(players); Iterator_next(players)) {
+            Player* const player = (Player*) Iterator_get(players);
+            assert(player);
+
+            if ( !Player_isNeutral(player) ) {
+                BwString* player_name = Player_getName(player);
```
Comment (0x7CFE) [inline example/Client.c]: `const`

File: example/Client.c [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                Race player_race = Player_getRace(player);
```
Comment (0x7CFE) [inline example/Client.c]: `const`

File: example/Client.c [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                Game_sendText(BWGame, "%s, playing as %d!", BwString_data(player_name), player_race.id);
```
Comment (0x7CFE) [inline example/Client.c]: `'%s'`

File: example/Client.c [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+                BwString_release(player_name);
+            }
+        }
+        Iterator_release(players);
+    }
+
+    while(Game_isInGame(BWGame))
+    {
+        Position text_position = { .x = 300, .y = 0 };
```
Comment (0x7CFE) [inline example/Client.c]: `const`

File: include/Game.h [hunk 8]
```diff
@@ -9,8 +9,8 @@ extern "C" {
 #endif
 
 /// Pass pointer from gameInit to initialize BWAPI::BroodwarPtr
```
Comment (0x7CFE) [inline include/Game.h]: Better append something like "`setGame()` must be called as part of the client init"

File: example/Client.c (hunk 1 above)
Comment (kpp) [inline example/Client.c]: statics are `0` by default

**Review by 0x7CFE — APPROVED** [review]

**Pull request merged**